Make October 2024 news public (#28954)
idaholab/moose: modules/doc/content/newsletter/2024/2024_10.md

- from the checkpoint file specified to the `Problem` for the restart
- from the default value of the initialization parameters, if and only if none of the previous conditions apply

### Direct Nodeset generation

In MOOSE, we often forego the concepts of nodesets and sidesets to simply use the term "boundary", which refers
to one or the other depending on the objects at hand. For example, a Dirichlet boundary condition for a Lagrange variable
will use a nodeset for its boundary parameter, while a postprocessor computing the diffusive flux with a monomial variable
will use a sideset for its boundary parameter. As such, we focused our mesh generation effort to be able to easily create sidesets,
and the nodesets can be easily generated from these sidesets.

However, for the specific need of generating 1D nodesets within a 3D mesh, we needed to be able to generate nodesets directly.
The [ParsedGenerateNodeset.md] joins the roster of mesh generators for that purpose. It offers much of the flexibility of
the [ParsedGenerateSideset.md] and their design are very similar.

> ### Mesh conversion to all simplices (triangles and tets)

Reviewer: @roystgnr good enough?

Author: Ping @roystgnr

Reviewer 2: Looks good to me; thanks very much for adding it.

The [ElementsToSimplicesConverter.md] was created as a mesh generator. An input mesh will be modified
to replace each non-simplex mesh element with a set of simplices
connecting the same nodes.  Each quad is split into 2 triangles, each
pyramid into 2 tetrahedra, each prism into 3 tets, and each cube into
6 tets.

> ### Ability to set multiple time integrators

Reviewer: @lindsayad  good enough?

Reviewer 3: good now

> For simulations using multiple equations, notably in the context of solving multiple nonlinear
> systems, there is now the option to use different time integration methods for each equation.

Reviewer 3: ```suggestion
For simulations using multiple variables/equations there is now the option to use different time integration methods for each variable/equation.
```

> ### New tests scheduler for High Performance Computing

Reviewer: @loganharbour  good enough?

A new scheduler was added to the MOOSE test suite to be able to leverage HPC platforms for
continuous integration.

## MOOSE Modules changes

### Navier Stokes: Skewness correction changes

The gradient of the pressure, both corrected and uncorrected, that are used in Rhie-Chow interpolation are now both
using skewness correction when skewness correction has been specified for the pressure variable.

Skewness-corrected gradients are now computed using the Least-Squares method rather than Green Gauss.
With these two modifications, oscillations on skewed meshes in simulations with volumetric body forces are no longer present.

> ## Thermal hydraulics: use of field variables on node elements for Junctions

Reviewer: @joshuahansel  good enough?
